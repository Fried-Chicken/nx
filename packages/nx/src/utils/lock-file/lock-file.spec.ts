import { mapLockFileDataToPartialGraph } from './lock-file';
import { parseNpmLockFile } from './npm';
import { parsePnpmLockFile } from './pnpm';
import { parseYarnLockFile } from './yarn';
import {
  lockFileYargsOnly as npmLockFileYargsOnly,
  lockFileV1YargsOnly as npmLockFileV1YargsOnly,
  lockFile as npmLockFile,
  lockFileV1 as npmLockFileV1,
} from './__fixtures__/npm.lock';
import {
  lockFileYargsOnly as pnpmLockFileYargsOnly,
  lockFile as pnpmLockFile,
} from './__fixtures__/pnpm.lock';
import {
  lockFileDevkitAndYargs,
  lockFile as yarnLockFile,
} from './__fixtures__/yarn.lock';

describe('lock-file', () => {
  describe('mapLockFileDataToExternalNodes', () => {
    it('should map yarn lock file data to external nodes', () => {
      const lockFileData = parseYarnLockFile(lockFileDevkitAndYargs);

      const partialGraph = mapLockFileDataToPartialGraph(lockFileData);

      expect(partialGraph.externalNodes['npm:yargs']).toMatchSnapshot();
      expect(partialGraph.dependencies['npm:yargs']).toMatchSnapshot();
    });

    it('should map successfully complex yarn lock file', () => {
      const lockFileData = parseYarnLockFile(yarnLockFile);

      const partialGraph = mapLockFileDataToPartialGraph(lockFileData);

      expect(partialGraph.externalNodes['npm:nx']).toMatchSnapshot();
      expect(partialGraph.dependencies['npm:nx']).toMatchSnapshot();
    });

    it('should map npm lock file data to external nodes', () => {
      const lockFileData = parseNpmLockFile(npmLockFileYargsOnly);

      const partialGraph = mapLockFileDataToPartialGraph(lockFileData);

      expect(partialGraph.externalNodes['npm:yargs']).toMatchSnapshot();
      expect(partialGraph.dependencies['npm:yargs']).toMatchSnapshot();
    });

    it('should map npm lock file v1 data to external nodes', () => {
      const lockFileData = parseNpmLockFile(npmLockFileV1YargsOnly);

      const partialGraph = mapLockFileDataToPartialGraph(lockFileData);

      expect(partialGraph.externalNodes['npm:yargs']).toMatchSnapshot();
      expect(partialGraph.dependencies['npm:yargs']).toMatchSnapshot();
    });

    it('should map successfully complex npm lock file', () => {
      const lockFileData = parseNpmLockFile(npmLockFile);

      const partialGraph = mapLockFileDataToPartialGraph(lockFileData);

      expect(partialGraph.externalNodes['npm:nx']).toMatchSnapshot();
      expect(partialGraph.dependencies['npm:nx']).toMatchSnapshot();
    });

    it('should map successfully complex npm v1 lock file', () => {
      const lockFileData = parseNpmLockFile(npmLockFileV1);

      const partialGraph = mapLockFileDataToPartialGraph(lockFileData);

      expect(partialGraph.externalNodes['npm:nx']).toMatchSnapshot();
      expect(partialGraph.dependencies['npm:nx']).toMatchSnapshot();
    });

    it('should map pnpm lock file data to external nodes', () => {
      const lockFileData = parsePnpmLockFile(pnpmLockFileYargsOnly);

      const partialGraph = mapLockFileDataToPartialGraph(lockFileData);

      expect(partialGraph.externalNodes['npm:yargs']).toMatchSnapshot();
      expect(partialGraph.dependencies['npm:yargs']).toMatchSnapshot();
    });

    it('should map successfully complex pnpm lock file', () => {
      const lockFileData = parsePnpmLockFile(pnpmLockFile);

      const partialGraph = mapLockFileDataToPartialGraph(lockFileData);

      expect(partialGraph.externalNodes['npm:nx']).toMatchSnapshot();
      expect(partialGraph.dependencies['npm:nx']).toMatchSnapshot();
      expect(
        partialGraph.externalNodes['npm:@phenomnomnominal/tsquery']
      ).toMatchSnapshot();
      expect(
        partialGraph.dependencies['npm:@phenomnomnominal/tsquery']
      ).toMatchSnapshot();
    });
  });
});
