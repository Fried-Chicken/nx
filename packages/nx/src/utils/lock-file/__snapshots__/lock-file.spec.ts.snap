// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lock-file mapLockFileDataToExternalNodes should map npm lock file data to external nodes 1`] = `
Object {
  "data": Object {
    "hash": "54342be9f4609650eba8e995881c5eb98c29b9d7bb95b9a699d9e9df68d6428e",
    "packageName": "yargs",
    "version": "17.5.1",
  },
  "name": "npm:yargs",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes should map npm lock file data to external nodes 2`] = `
Array [
  Object {
    "source": "npm:yargs",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:escalade",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:get-caller-file",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:require-directory",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:y18n",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes should map npm lock file v1 data to external nodes 1`] = `
Object {
  "data": Object {
    "hash": "c22918aa52c9fdce78bf82451d8114e00b6d4375a8e5b1d1d96c6cf4755dbe1a",
    "packageName": "yargs",
    "version": "17.5.1",
  },
  "name": "npm:yargs",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes should map npm lock file v1 data to external nodes 2`] = `
Array [
  Object {
    "source": "npm:yargs",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:escalade",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:get-caller-file",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:require-directory",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:y18n",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes should map pnpm lock file data to external nodes 1`] = `
Object {
  "data": Object {
    "hash": "c22918aa52c9fdce78bf82451d8114e00b6d4375a8e5b1d1d96c6cf4755dbe1a",
    "packageName": "yargs",
    "version": "17.5.1",
  },
  "name": "npm:yargs",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes should map pnpm lock file data to external nodes 2`] = `
Array [
  Object {
    "source": "npm:yargs",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:escalade",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:get-caller-file",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:require-directory",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:y18n",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes should map successfully complex npm lock file 1`] = `
Object {
  "data": Object {
    "hash": "a20b2870460e95ffeffa1fa07d8430a61b225743644f5646c8dd276124683d7b",
    "packageName": "nx",
    "version": "14.7.5",
  },
  "name": "npm:nx",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes should map successfully complex npm lock file 2`] = `
Array [
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/cli",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/tao",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@parcel/watcher",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chalk",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chokidar",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-cursor",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-spinners",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:dotenv",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:enquirer",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fast-glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:figures",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:flat",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fs-extra",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:ignore",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:js-yaml@4.1.0",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:jsonc-parser",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:minimatch",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:npm-run-path",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:open",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:semver",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tar-stream",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tmp",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tsconfig-paths",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tslib",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:v8-compile-cache",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes should map successfully complex npm v1 lock file 1`] = `
Object {
  "data": Object {
    "hash": "3e63cd88ec8f5eb00f37559c24caf47da1f45f6a3f607d013f148da8593e3a26",
    "packageName": "nx",
    "version": "14.7.5",
  },
  "name": "npm:nx",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes should map successfully complex npm v1 lock file 2`] = `
Array [
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/cli",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/tao",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@parcel/watcher",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chalk",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chokidar",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-cursor",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-spinners",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:dotenv",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:enquirer",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fast-glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:figures",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:flat",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fs-extra",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:ignore",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:js-yaml",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:jsonc-parser",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:minimatch",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:npm-run-path",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:open",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:semver",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tar-stream",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tmp",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tsconfig-paths",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tslib",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:v8-compile-cache",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes should map successfully complex pnpm lock file 1`] = `
Object {
  "data": Object {
    "hash": "1e61b9db38c17534ce427d40ce84f9c03b4c439253f659e9a48513a10a52b465",
    "packageName": "nx",
    "version": "14.7.5",
  },
  "name": "npm:nx",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes should map successfully complex pnpm lock file 2`] = `
Array [
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/cli",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/tao",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@parcel/watcher",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chalk",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chokidar",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-cursor",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-spinners",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:dotenv",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:enquirer",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fast-glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:figures",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:flat",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fs-extra",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:ignore",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:js-yaml",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:jsonc-parser",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:minimatch",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:npm-run-path",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:open",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:semver",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tar-stream",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tmp",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tsconfig-paths",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tslib",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:v8-compile-cache",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes should map successfully complex pnpm lock file 3`] = `
Object {
  "data": Object {
    "hash": "6c5961e1a6ab89174727587860e480508f945407ce4c2cc7c9f1e46f181e6468",
    "packageName": "@phenomnomnominal/tsquery",
    "version": "4.1.1",
  },
  "name": "npm:@phenomnomnominal/tsquery",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes should map successfully complex pnpm lock file 4`] = `
Array [
  Object {
    "source": "npm:@phenomnomnominal/tsquery",
    "target": "npm:esquery",
    "type": "static",
  },
  Object {
    "source": "npm:@phenomnomnominal/tsquery",
    "target": "npm:typescript",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes should map successfully complex yarn lock file 1`] = `
Object {
  "data": Object {
    "hash": "1e61b9db38c17534ce427d40ce84f9c03b4c439253f659e9a48513a10a52b465",
    "packageName": "nx",
    "version": "14.7.5",
  },
  "name": "npm:nx",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes should map successfully complex yarn lock file 2`] = `
Array [
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/cli",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@nrwl/tao",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:@parcel/watcher",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chalk",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:chokidar",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-cursor",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cli-spinners",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:dotenv",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:enquirer",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fast-glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:figures",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:flat",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:fs-extra",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:glob",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:ignore",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:js-yaml",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:jsonc-parser",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:minimatch",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:npm-run-path",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:open",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:semver",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tar-stream",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tmp",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tsconfig-paths",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:tslib",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:v8-compile-cache",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs",
    "type": "static",
  },
  Object {
    "source": "npm:nx",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;

exports[`lock-file mapLockFileDataToExternalNodes should map yarn lock file data to external nodes 1`] = `
Object {
  "data": Object {
    "hash": "c22918aa52c9fdce78bf82451d8114e00b6d4375a8e5b1d1d96c6cf4755dbe1a",
    "packageName": "yargs",
    "version": "17.5.1",
  },
  "name": "npm:yargs",
  "type": "npm",
}
`;

exports[`lock-file mapLockFileDataToExternalNodes should map yarn lock file data to external nodes 2`] = `
Array [
  Object {
    "source": "npm:yargs",
    "target": "npm:cliui",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:escalade",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:get-caller-file",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:require-directory",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:string-width",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:y18n",
    "type": "static",
  },
  Object {
    "source": "npm:yargs",
    "target": "npm:yargs-parser",
    "type": "static",
  },
]
`;
